include/arch/x86/intel64/thread.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_X86_INTEL64_THREAD_H_
+#define ZEPHYR_INCLUDE_ARCH_X86_INTEL64_THREAD_H_
+
+#define X86_THREAD_FLAG_ALL 0x01 /* _thread_arch.flags: entire state saved */
+
+/*
+ * GDT selectors - these must agree with the GDT layout in locore.S.
+ */
+
+#define X86_KERNEL_CS_32	0x08	/* 32-bit kernel code */
+#define X86_KERNEL_DS_32	0x10	/* 32-bit kernel data */
+#define X86_KERNEL_CS		0x18	/* 64-bit kernel code */
+#define X86_KERNEL_DS		0x20	/* 64-bit kernel data */
+
+#define X86_KERNEL_CPU0_GS	0x30	/* data selector covering TSS */
+#define X86_KERNEL_CPU0_TR	0x40	/* 64-bit task state segment */
+#define X86_KERNEL_CPU1_GS	0x50	/* data selector covering TSS */
+#define X86_KERNEL_CPU1_TR	0x60	/* 64-bit task state segment */
+#define X86_KERNEL_CPU2_GS	0x70	/* data selector covering TSS */
+#define X86_KERNEL_CPU2_TR	0x80	/* 64-bit task state segment */
+#define X86_KERNEL_CPU3_GS	0x90	/* data selector covering TSS */
+#define X86_KERNEL_CPU3_TR	0xA0	/* 64-bit task state segment */
+
+/*
+ * Some SSE definitions. Ideally these will ultimately be shared with 32-bit.
+ */
+
+#define X86_FXSAVE_SIZE		512	/* size and alignment of buffer ... */
+#define X86_FXSAVE_ALIGN	16	/* ... for FXSAVE/FXRSTOR ops */
+#define X86_MXCSR_SANE		0x1dc0	/* enable division-by-zero exception */
+
+#ifndef _ASMLANGUAGE
+
+#include <zephyr/types.h>
+
+/*
+ * 64-bit Task State Segment. One defined per CPU.
+ */
+
+struct x86_tss64 {
```

Comment:
this could really go in arch/x86/include somewhere

all we really need in this file is things necessary to complete the thread struct

Reply:
If I remember correctly, it was in a private header, then something in public needed this so I moved it to thread.h. I will check again.

Reply:
The reason I moved `x86_tss64` (`x86_tss64_t`) from the private `kernel_data_arch.h` to the public `thread.h` is that x86_64 `z_arch_curr_cpu` requires it. Note that `z_arch_curr_cpu` is used by `_current` in `kernel_structs.h`, which is used by public inline functions (for instance, in `sys_mutex_unlock` in `sys/mutex.h`.